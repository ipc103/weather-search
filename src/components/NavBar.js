import React from 'react'

const NavBar = ({brand}) => {
  return (
    <nav className="navbar navbar-default">
      <div className='navbar-header'>
        <a className="navbar-brand">{brand}</a>
      </div>
    </nav>
  )
}

export default NavBar
