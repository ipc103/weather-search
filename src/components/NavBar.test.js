import React from 'react'
import {shallow} from 'enzyme'
import NavBar from './NavBar'

describe('NavBar', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<NavBar/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('renders with a brand link', () => {
    const wrapper = shallow(<NavBar brand="Coffee" />)
    expect(wrapper.find('.navbar-brand').text()).toEqual('Coffee')
  })
})
