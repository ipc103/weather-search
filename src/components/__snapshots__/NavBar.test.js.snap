// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <nav
    className="navbar navbar-default"
>
    <div
        className="navbar-header"
    >
        <a
            className="navbar-brand"
        />
    </div>
</nav>,
  "nodes": Array [
    <nav
      className="navbar navbar-default"
>
      <div
            className="navbar-header"
      >
            <a
                  className="navbar-brand"
            />
      </div>
</nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav
          className="navbar navbar-default"
>
          <div
                    className="navbar-header"
          >
                    <a
                              className="navbar-brand"
                    />
          </div>
</nav>,
        "_debugID": 2,
        "_renderedOutput": <nav
          className="navbar navbar-default"
>
          <div
                    className="navbar-header"
          >
                    <a
                              className="navbar-brand"
                    />
          </div>
</nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
